### PR TITLE
Color input in visual tests

### LINKED ISSUE #163: Color input in visual tests
Add a nicer color input instead of having to type (255, 255, 255) into the command panel yikes

This story should just focus on getting the color input into our visual tests. We have #448 to actually implement it.

Some people have started some work on their own input: https://github.com/ppy/osu-framework/issues/3134

A few Alex has looked at:
* https://github.com/vignette-project/vignette/pull/161
* https://github.com/Flutterish/Hitokori/releases/tag/0.18.4-2021.03.15
* https://github.com/ppy/osu-framework/pull/3138

### PR DESCRIPTION
**Pull request by maxrchung**
Closes #163

### CODE REVIEW

**Comment by Xenocidel** [comment]
Pushed example usage of the colorPicker to the visual test. Basically you BindValueChanged and send `value.newValue` off to wherever you need it
```
S2VXColorPicker colorPicker;
SpriteText colorText;
Add(colorPicker = new S2VXColorPicker { Margin = new MarginPadding(20) });
Add(colorText = new SpriteText { });
colorPicker.Current.BindValueChanged(value => colorText.Text = value.NewValue.ToString());
```

**Review by Xenocidel — APPROVED** [review]

File: S2VX.Game/Editor/ColorPicker/S2VXColorPicker.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+using osuTK;
+using osuTK.Graphics;
+
+namespace S2VX.Game.Editor.ColorPicker {
+    public class S2VXColorPicker : Container, IHasCurrentValue<Color4> {
+        // Change current value will cause recursive change, so need a record to disable this change.
+        private readonly Cached InternalUpdate = new Cached();
+
+        private readonly BindableWithCurrent<Color4> CurrentColor =
+            new BindableWithCurrent<Color4> { Default = Color4.White };
```
Comment (Naranja-Sagged) [inline S2VX.Game/Editor/ColorPicker/S2VXColorPicker.cs:19]: Can probably be simplified to new() to follow C# 9

Comment (Xenocidel) [reply S2VX.Game/Editor/ColorPicker/S2VXColorPicker.cs]: Implemented. Also did the same for attributes in other files

**Comment by Xenocidel** [comment]
I noticed that the color picker starts out with a value of #000000 or RGBA (0, 0, 0, 1), but the red dot is at the top left which corresponds to White.

File: S2VX.Game/Editor/ColorPicker/S2VXColorPicker.cs (hunk 1 above)
Comment (Naranja-Sagged) [reply S2VX.Game/Editor/ColorPicker/S2VXColorPicker.cs]: i think you missed the one on line 16 lol

Comment (Xenocidel) [reply S2VX.Game/Editor/ColorPicker/S2VXColorPicker.cs]: Whoops. Fixed.

**Review by Naranja-Sagged — APPROVED** [review]

**Comment by maxrchung** [comment]
Gonna merge this through with a change you guys should probably be aware about. While doing some more testing, I noticed that the hue slider was going the wrong direction. E.g. hue slider was displaying red -> blue -> green -> red when in reality it was red -> green -> blue -> red. I flipped this around in this commit: https://github.com/maxrchung/S2VX/pull/451/commits/b975063476a4dd31b088f3412ad3605f2994edfa

**Pull request merged**